tools/downloader/caffe2_to_onnx.py

```diff
@@ -0,0 +1,92 @@
+import argparse
+from pathlib import Path
+import sys
+import os
+import re
+
+import onnx
+from caffe2.python.onnx.frontend import Caffe2Frontend
+from caffe2.proto import caffe2_pb2
+
+
+def positive_int_arg(values):
+    """Check positive integer type for input argument"""
+    result = []
+    shapes = re.findall(r'[(\[]([0-9, -]+)[)\]]', values)
+    for shape in shapes:
+        single_shape = []
+        for value in shape.split(','):
+            try:
+                ivalue = int(value)
+                if ivalue < 0:
+                    raise argparse.ArgumentTypeError('Argument must be a positive integer')
+                single_shape.append(ivalue)
+            except Exception as exc:
+                print(exc)
+                sys.exit('Invalid value for input argument: {!r}, a positive integer is expected'.format(value))
+        result.append(single_shape)
+    return result
+
+def parse_args():
+    """Parse input arguments"""
+
+    parser = argparse.ArgumentParser(description='Conversion of pretrained models from Caffe2 to ONNX')
+
+    parser.add_argument('--model-name', type=str, required=True,
+                        help='Model to convert. May be class name or name of constructor function')
+    parser.add_argument('--output-file', type=Path, required=True,
+                        help='Path to the output ONNX model')
+    parser.add_argument('--model-path', type=str, required=True,
+                        help='Path to predict_net .pb file')
+    parser.add_argument('--weights', type=str, required=True,
+                        help='Path to init_net .pb file')
+    parser.add_argument('--input-shape', metavar='INPUT_DIM', type=positive_int_arg,
+                        required=True, help='Shape of the input blob')
+    parser.add_argument('--input-names', type=str,
+                        help='Comma separated names of the input layers')
+
+    return parser.parse_args()
+
+def convert_to_onnx(predict_net_path, init_net_path, input_shape, input_names, output_file, model_name=''):
+    """Convert Caffe2 model to ONNX and check the resulting onnx model"""
+
+    output_file.parent.mkdir(parents=True, exist_ok=True)
+
+    value_info = {}
+    data_type = onnx.TensorProto.FLOAT
+    input_names = input_names.split(',')
+    for name, shape in zip(input_names, input_shape):
+        value_info[name] = [data_type, shape]
+
+    predict_net = caffe2_pb2.NetDef()
+    with open(predict_net_path, 'rb') as file:
```

Comment:
It would be simpler to replace this with `predict_net_path.read_bytes()`. Likewise for other `open`s in this function.